rechunker/types.py

```diff
@@ -0,0 +1,20 @@
+from typing import Any, Iterable, NamedTuple, Tuple
+
+import zarr
+
+
+class CopySpec(NamedTuple):
+    """Specifcation of how to copy between two arrays."""
+
+    # TODO: remove Any by making CopySpec a Generic, once we only support Python
+    # 3.7+: https://stackoverflow.com/questions/50530959
+    source: Any
+    target: zarr.Array
+    chunks: Tuple[int, ...]
+
+
+class StagedCopySpec:
+    stages: Tuple[CopySpec, ...]
+
+    def __init__(self, stages: Iterable[CopySpec]):
+        self.stages = tuple(stages)
```

Comment:
We might consider adding (or at least documenting) some invariant checks when constructing these objects, e.g.,

- `source` and `target` should have same `shape` and `dtype`
- no element of `chunks` should be larger than `shape`
- the `target` and `source` of subsequent stages should be the same array